pallets/rc-migrator/src/scheduler.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use frame_support::traits::schedule::v3::TaskName;
+use pallet_scheduler::{RetryConfig, TaskAddress};
+
+/// Stage of the scheduler pallet migration.
+#[derive(Encode, Decode, Clone, Default, RuntimeDebug, TypeInfo, MaxEncodedLen, PartialEq, Eq)]
+pub enum SchedulerStage<BlockNumber> {
+	#[default]
+	IncompleteSince,
+	Agenda(Option<BlockNumber>),
+	Retries(Option<TaskAddress<BlockNumber>>),
+	Lookup(Option<TaskName>),
+	Finished,
+}
+
+/// Message that is being sent to the AH Migrator.
+#[derive(Encode, Decode, Debug, Clone, TypeInfo, MaxEncodedLen, PartialEq, Eq)]
+pub enum RcSchedulerMessage<BlockNumber, Scheduled> {
+	IncompleteSince(BlockNumber),
+	Agenda((BlockNumber, Vec<Option<Scheduled>>)),
+	Retries((TaskAddress<BlockNumber>, RetryConfig<BlockNumber>)),
+	Lookup((TaskName, TaskAddress<BlockNumber>)),
+}
+
+pub type RcSchedulerMessageOf<T> = RcSchedulerMessage<BlockNumberFor<T>, alias::ScheduledOf<T>>;
+
+pub struct SchedulerMigrator<T: Config> {
+	_phantom: PhantomData<T>,
+}
+
+impl<T: Config> PalletMigration for SchedulerMigrator<T> {
+	type Key = SchedulerStage<BlockNumberFor<T>>;
+	type Error = Error<T>;
+	fn migrate_many(
+		last_key: Option<Self::Key>,
+		weight_counter: &mut WeightMeter,
+	) -> Result<Option<Self::Key>, Self::Error> {
+		let mut last_key = last_key.unwrap_or(SchedulerStage::IncompleteSince);
+		let mut messages = Vec::new();
+
+		loop {
+			if weight_counter
+				.try_consume(<T as frame_system::Config>::DbWeight::get().reads_writes(1, 1))
+				.is_err()
+			{
+				if messages.is_empty() {
+					return Err(Error::OutOfWeight);
+				} else {
+					break;
+				}
+			}
+			if messages.len() > 10_000 {
+				log::warn!("Weight allowed very big batch, stopping");
+				break;
+			}
+
+			last_key = match last_key {
+				SchedulerStage::IncompleteSince => {
+					if let Some(since) = pallet_scheduler::IncompleteSince::<T>::take() {
+						messages.push(RcSchedulerMessage::IncompleteSince(since));
+					}
+					SchedulerStage::Agenda(None)
+				},
+				SchedulerStage::Agenda(last_key) => {
+					let mut iter = if let Some(last_key) = last_key {
+						alias::Agenda::<T>::iter_from_key(last_key)
```

Comment:
Not sure if you have to call `hashed_key_for`. Did you try it it migrates all?

Reply:
I think so, I will double check. I used `iter_from_key` for all other pallets migrations

Reply:
Yea but it is tricky since it sometimes does still catch a subset. We will see when testing,  it could be a likely culprit when entries are missing.